webpack/components/AnsibleRolesSwitcher/AnsibleRolesSwitcherReducer.js

```diff
@@ -35,22 +37,16 @@ const initAssignedRoles = (state, initialAssignedRoles, inheritedRoleIds) => {
         role
     ));
 
-    return state.set('assignedRoles', assignedRoles).set('initialized', true);
+    return state.merge({ assignedRoles, initialized: true });
   }
   return state;
 };
 
 const ansibleRoleAdd = (state, payload) =>
-  state.set('assignedRoles', addItem(state.assignedRoles, payload.role))
-       .set('itemCount', state.itemCount - 1);
+  state.merge(addItemNewState(state, payload.role));
 
 const ansibleRoleRemove = (state, payload) =>
-  state.set('assignedRoles', removeItem(state.assignedRoles, payload.role))
-       .set('itemCount', state.itemCount + 1);
-
-const addItem = (list, item) => ([...(list || []), item]);
-
-const removeItem = (list, item) => list.filter(listItem => item.id !== listItem.id);
+  state.merge(removeItemNewState(state, payload.role));
```

Comment:
hmm.. sorry, maybe I wasn't clear, the idea was to consider using reselect instead of storing it in the store? /cc @sharvit